File: object-oriented-programming/final-game/python/card_class.py
# Isaac Computer Science
# Usage licensed under the Open Government Licence v3.0

class Card:
    """Defines a standard playing card"""

    def __init__(self, given_suit, given_rank, given_value):
        self.__suit = given_suit
        self.__rank = given_rank
        self.__value = given_value


    def get_suit(self):
        return self.__suit

    def get_rank(self):
        return self.__rank

    def get_value(self):
        return self.__value

    def __repr__(self):
        return(f"{self.__rank} of {self.__suit}")
    

# This code will run if this file is executed directly
# (i.e. not called by another program)

if __name__ == '__main__':        
    two_of_clubs = Card("clubs", "2", 2) # Instantiate an example Card object

File: object-oriented-programming/fundamentals/c-sharp/card_class.cs
/*
Isaac Computer Science
Usage licensed under the Open Government Licence v3.0

Note: This file is designed to be copied out and compiled on your machine.
In order for it to compile properly you need to ensure that the project name is the same as the "namespace" in this file.
To run this file you need to:
1. Copy the contents
2. Paste them into the C# IDE of your choice (Visual Studio, for example)
3. Change the namespace to match your project (if necessary)
4. Compile the program
5. Run the program
*/

using System;
using System.IO;

namespace IsaacCodeSamples
{
    class Card
    {

        private string suit;
        private string rank;
        private int value;

        // Constructor method
        public Card(string givenSuit, string givenRank, int givenValue) {
            suit = givenSuit;
            rank = givenRank;
            value = givenValue;
        }

        
    }
    
    
    class Testing
    {
        // The Main method is the entry point for all C# programs
        public static void Main()
        {
            // Instantiate a new card object
            Card two_of_clubs = new Card("clubs", "2", 2);
        }
    }
    
    
}
